src/bp/core/modules/require.ts

```diff
@@ -98,3 +98,20 @@ export const buildLookupPaths = (module: string, locations: string[]) => {
     })
   )
 }
+
+export const clearModuleScriptCache = (moduleLocation: string, depth: number = 0) => {
+  const cacheKey = require.resolve(moduleLocation)
+  const file = require.cache[cacheKey]
+
+  if (file) {
+    for (const { filename } of file.children) {
+      // Circular reference protection, we only unload the user's module files
+      if (depth < 3 && !filename.includes('node_modules')) {
```

Comment:
What does the `3` stand for in this context?

Reply:
I've clarified the magic number. It will clear index.js, then will check to clear its dependencies (level 1), then check the dependencies of that dependency (lvl 2), and then one last time. 

This limit is required to prevent circular references, in case one file references another, which references the first one again

Reply:
Thanks!